### PR TITLE
adding 3 questions including find-dup-num

### PR DESCRIPTION
**Pull request by roastedbill**

### CODE REVIEW

File: algorithms/longest non-repeat substring/Solution.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/**
```
Comment (franklingu) [inline algorithms/longest non-repeat substring/Solution.java]: Is this file naming not the same with the rest?
like "longest-substring-without-repeating-characters"
I would suggest we do exact matching for url

**Comment by franklingu** [comment]
maybe can also modify the Readme for this patch.
and pls follow that format(i know, it is troublesome, but it is clearer mapping) thanks

**Comment by franklingu** [comment]
when you are done with these small fixes, just push. but I do suggest you have another branch instead of master in the future.

**Comment by roastedbill** [comment]
I've updated folder names as well as readme.md

**Comment by franklingu** [comment]
@roastedbill , if you are busy, then never mind. if not and you want to learn something about PR, you can read about [Guide for adding remote](https://help.github.com/articles/adding-a-remote/). Add this repo as upstream and pull from upstream to merge it--when you find conflicts, resolve it and re-push

Some other refs:
[Guide for syncing with fork](https://help.github.com/articles/syncing-a-fork/)
[Another post](https://help.github.com/articles/merging-an-upstream-repository-into-your-fork/)

**Comment by franklingu** [comment]
If you are busy, just leave it and I will merge tonight when I am free

**Pull request merged**